### ISSUE TITLE
Remove notifications for friends who have gone back offline

### ISSUE THREAD

**Issue by elidourado** [devonzuegel/ketchup-club#8]
Scenario: User receives a notification saying that a friend is online, but doesn't see it for some period of time. In that period of time, the friend goes back offline. The user is excited to call his friend and does so once he sees the notification. The friend is no longer available, so this is not a good user experience for anyone.

We can remove past notifications when friends go offline with Expo's [dismissNotificationAsync](https://docs.expo.dev/versions/latest/sdk/notifications/#dismissnotificationasyncnotificationidentifier) method.

**Comment by elidourado** [comment]
Note that this will have to happen in the background.

**Comment by devonzuegel** [comment]
I agree that this is a problem as it's currently designed. I'm not sure if removing the notification is the right solution, though I also see value in it. Let's explore a few solutions before we land on one.

I had been planning to do something more like: when you click the notification, it sends you to the home screen which gives you a little message "Eli was online until 5 mins ago". The reason I like that is because if they just went offline, they may actually still be available so you might want to give them a call anyways.

Another way I was thinking about it was to have a running log of when people are online/offline, so you can see history and get context of when people tend to go online. That might be weird though.

Overall, your solution is maybe simpler and more elegant than either of these ideas though. I'm not married to one or the other just yet. I want to throw it together in a Figma flowchart or prototype to get a feel for which one feels more natural. Wdyt?

**Comment by elidourado** [comment]
Totally agree that there could be a better solution, and it's a good idea to mock up different ideas.

One use case to be concerned about would be if I see a "Devon is online" notification and I don't click the notification at all, I just say "Siri, call Devon." Then I'll miss the screen that says you just went offline.

**Comment by devonzuegel** [comment]
Ahh I see. Initially KC sent a notification when someone went offline too, but that ended up feeling noisy so I removed it. But now that I think of it, it might have just felt noisy because I was doing a ton of testing at the time, and with a more normal level of usage it wouldn't feel so noisy. Maybe we should experiment with putting it back in for now as a stopgap.
